app/Http/Controllers/Admin/AdminController.php

```diff
@@ -15,21 +39,145 @@ class AdminController extends Controller
      */
     public function index()
     {
-        return view('admin.index');
+        return view('admin.user.dashboard');
     }
 
+    /**
+     * Log account admin out of the application.
+     *
+     * @return void
+     */
     public function userList()
     {
-        return view('admin.user.list');
+        $users = Auth::user();
+        if ($users->role == \App\User::ADMIN) {
+            $users = User::where('role', 2)->paginate(3);
+            return view('admin.user.list', ['users' => $users]);
+        } else {
+            return redirect()->back();
+        }
     }
 
+    /**
+     * Log account admin out of the application.
+     *
+     * @return void
+     */
     public function userCreate()
     {
         return view('admin.user.createOrUpdate');
     }
 
-    public function userEdit()
+    /**
+     * Log account admin out of the application.
+     *
+     * @param Request $request Request create user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function handleUserCreate(Request $request)
     {
-        return view('admin.user.createOrUpdate');
+        $this->validate(
+            $request, [
+            'username' => 'required',
+            'email' => 'required',
+            'password' => 'required', 
+            ], [ 
+            'username.required' => 'Username required',
+            'email.required' => 'Email required',
+            'password.required' => 'Password required', 
```

Comment:
this is message custom error... it had message default...can you see in resource/lang/validation.

Reply:
ok, i used message default